docs/perfect-pull-request.md

```diff
@@ -0,0 +1,229 @@
+    number.
+
+7.  Save the new commit message, check it looks ok in the **git log**
+
+8.  If you're squashing after initiating the pull request, you will
+    need to `git push --force` when pushing the commit back to github.
+
+## Updating your Pull Request after changes are made<a id="sec-1-7" name="sec-1-7"></a>
+
+You can make changes after your pull request is submitted,
+effectively it is still private branch. If it's necessary to
+add new code, to add a new test, or fix something.
+
+Add all the changes interactively using `git add -p` and
+then `git commit --amend` to patch the previous commit with
+your amend.
+
+Please note, you can also add your changes as new commits,
+and then squash them before pushing the changes up to github
+
+When you push your amended (or newly squashed) commits, you
+must use the `--force` switch. ie. `git push --force`
+
+## Simple vs Complex<a id="sec-1-8" name="sec-1-8"></a>
+
+If the piece of work you are submitting is complex, and
+touches many parts of the system, you should separate the
+work into a set of commits, instead of a single one.
+
+If you are unsure about the level of complexity in your
+work, check other pull-request which have been accepted, how
+does your stack up?
+
+If you still can't decide, squash your commits into a group
+of logical sections of work, ensure that any dependencies
+are met within each group, and that they build / pass their tests.
+
+You can check this before squashing, by running the test
+suite on the last commit in a set.  Never send a broken
+commit in your pull requests.
+
+## When your Pull Request contains unwanted commits<a id="sec-1-9" name="sec-1-9"></a>
+
+If your pull-request has commits that are unwanted, you can
+either squash them out, or simply start a new branch, and
+use cherry pick to construct a new version of the branch,
+omitting the bad commit.  Note that rebase squash will
+usually take care of this situation.
+
+Git cherry pick, is fairly simple to use.  View the git log
+of your current branch, make a note of the commit SHA1s you
+want to keep.
+
+Next, create a new local branch, and cherry pick each
+commit into the new branch.
+
+e.g.  if we noted that commits, `823a69b`, `a98e12c` and `a9171ad` are
+good, and the rest are trash, we'd do.
+
+```
+git checkout -b new-branch-for-bounty-123
+# We're now in the new branch...
+
+for sha in 823a69b a98e12c a9171ad; do
+    git cherry-pick $sha
+done
+```
+
+We will need to open a new pull request for this branch. as
+mentioned above, an extra commit to tidy up followed by rebase 
+pick/squash may be able to take care of this within the existing
+branch. The only time you'd really want to do a cherry pick, is 
+when it's simpler to do so.
+
+If you have any questions about our pull request process,
+please contact the coderwall team.  One of us will be happy
+to help.
+
+## Fix those errors
+
+Found a factual mistake in this document? Let me know and I’ll correct it.
```

Comment:
instead of ' ̀Let me know and I’ll correct it' :  Or you can start applying what you just read in this doc and  submit a PR !